sources/models/shared-defaults/kubernetes-services.toml

```diff
@@ -40,7 +40,7 @@ path = "/etc/kubernetes/pki/kubelet-server.crt"
 template-path = "/usr/share/templates/kubelet-server-crt"
 
 [configuration-files.kubelet-server-key]
-path = "/etc/kubernetes/pki/kubelet-server.key"
+path = "/etc/kubernetes/pki/private/kubelet-server.key"
```

Comment:
@webern / @etungsten  isn't a migration required for this?

Reply:
This concerns the `configuration-files.kubelet-server-key` value in the settings datastore. These defaults are only applied upon first-boot after the settings datastore has been created and populated with defaults. When we change the default and we want all hosts to start to render a configuration file to a different `path` to fix some issue, we need to ensure existing host out there also use the correct `path` value here after upgrading. So we need a migration to check on upgrade if that old value is in the datastore, and if it is and we know it's incorrect, then we migrate it to the new value. 

We actually don't have an example migration of this kind, so you have to implement `Migration` trait. I can help you with that. Here's example migration you can reference: https://github.com/bottlerocket-os/bottlerocket/blob/3cbe1d843f3abcdc87efefd3679c7ece0378693e/sources/api/migration/migrations/archived/v1.1.0/shared-containerd-configs/src/main.rs

Instead of `template-path`, we're replacing the `path`.

Reply:
Added a migration. Please take a look and confirm that is what's needed when you get a chance. Thanks!